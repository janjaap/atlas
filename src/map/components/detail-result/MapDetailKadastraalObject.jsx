import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultItem from './MapDetailResultItem';
import MapDetailResultWrapper from './MapDetailResultWrapper';

const MapDetailKadastraalObject = ({
  panoUrl, kadastraalObject, onMaximize, onPanoPreviewClick
}) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={kadastraalObject.label}
    title="Kadastraal object"
  >
    <ul className="map-detail-result__list">
      {kadastraalObject.kadastraleGemeente && (
        <MapDetailResultItem
          label="Kadastrale gemeente"
          value={kadastraalObject.kadastraleGemeente.name}
        />
      )}
      <MapDetailResultItem
        label="Gemeente"
        value={kadastraalObject.kadastraleGemeente.gemeente}
      />
      <MapDetailResultItem
        label="Grootte"
        value={(kadastraalObject.size || kadastraalObject.size === 0) ? `${kadastraalObject.size.toLocaleString('nl-NL')} mÂ²` : ''}
      />
    </ul>
  </MapDetailResultWrapper>
);

MapDetailKadastraalObject.propTypes = {
  kadastraalObject: PropTypes.shape({
    kadastraleGemeente: PropTypes.shape({
      label: PropTypes.string,
      name: PropTypes.string,
      gemeente: PropTypes.string
    }),
    label: PropTypes.string,
    size: PropTypes.number
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailKadastraalObject;
