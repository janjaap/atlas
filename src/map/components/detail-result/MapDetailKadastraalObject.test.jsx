import React from 'react';
import { shallow } from 'enzyme';

import MapDetailKadastraalObject from './MapDetailKadastraalObject';

describe('MapDetailKadastraalObject', () => {
  it('should render everything', () => {
    const kadastraalObject = {
      kadastraleGemeente: {
        label: 'Kadastrale gemeente label',
        name: 'Kadastrale gemeente name'
      },
      label: 'Kadastraal object label',
      objectNumber: '123ABC',
      size: 115
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailKadastraalObject
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        kadastraalObject={kadastraalObject}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without kadastrale gemeente', () => {
    const kadastraalObject = {
      label: 'Kadastraal object label',
      objectNumber: '123ABC',
      size: 115
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailKadastraalObject
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        kadastraalObject={kadastraalObject}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with zero value for size', () => {
    const kadastraalObject = {
      label: 'Kadastraal object label',
      objectNumber: '123ABC',
      size: 0
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailKadastraalObject
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        kadastraalObject={kadastraalObject}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render size', () => {
    const kadastraalObject = {
      label: 'Kadastraal object label',
      objectNumber: '123ABC'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailKadastraalObject
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        kadastraalObject={kadastraalObject}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
