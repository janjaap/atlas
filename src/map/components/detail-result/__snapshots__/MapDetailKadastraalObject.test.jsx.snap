// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailKadastraalObject should not render size 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Kadastraal object label"
  title="Kadastraal object"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Kadastrale gemeente"
      value=""
    />
    <MapDetailResultItem
      label="Gemeente"
      value=""
    />
    <MapDetailResultItem
      label="Grootte"
      value=""
    />
  </ul>
</MapDetailResultWrapper>
`;

exports[`MapDetailKadastraalObject should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Kadastraal object label"
  title="Kadastraal object"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Kadastrale gemeente"
      value="Kadastrale gemeente name"
    />
    <MapDetailResultItem
      label="Gemeente"
      value="Gemeentenaam!"
    />
    <MapDetailResultItem
      label="Grootte"
      value="115 m²"
    />
  </ul>
</MapDetailResultWrapper>
`;

exports[`MapDetailKadastraalObject should render with zero value for size 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Kadastraal object label"
  title="Kadastraal object"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Kadastrale gemeente"
      value=""
    />
    <MapDetailResultItem
      label="Gemeente"
      value=""
    />
    <MapDetailResultItem
      label="Grootte"
      value="0 m²"
    />
  </ul>
</MapDetailResultWrapper>
`;

exports[`MapDetailKadastraalObject should render without kadastrale gemeente 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Kadastraal object label"
  title="Kadastraal object"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Kadastrale gemeente"
      value=""
    />
    <MapDetailResultItem
      label="Gemeente"
      value=""
    />
    <MapDetailResultItem
      label="Grootte"
      value="115 m²"
    />
  </ul>
</MapDetailResultWrapper>
`;
